src/main/java/com/bobocode/hoverla/bring/context/ApplicationContextImpl.java

```diff
@@ -1,60 +1,152 @@
-           2) Merge List of bean definitions into one List
-           3) Send beans list to validator to check duplicates
-           4) Convert bean list to map
-           5) Send beans map to initializer to instantiate beans and inject dependencies
-         */
+
+        List<BeanDefinition> beanDefinitionList = scanningApplication(scanners);
+        validate(validator, beanDefinitionList);
+        contextCreation(beanDefinitionList);
+        beansInstantiation(initializer);
+
+        log.info("Application context initialized");
+    }
+
+    private List<BeanDefinition> scanningApplication(List<BeanScanner> scanners) {
+        List<BeanDefinition> beanDefinitionList = new ArrayList<>();
+        log.info("Scanning application for finding bean definitions");
+        for (BeanScanner beanScanner : scanners) {
+            log.debug("Running bean scanner {}", beanScanner.getClass().getSimpleName());
+            beanDefinitionList.addAll(beanScanner.scan());
+        }
+        return beanDefinitionList;
+    }
+
+    private void validate(BeanValidator validator, List<BeanDefinition> beanDefinitionList) {
+        log.info("Beans definitions validating");
+        validator.validate(beanDefinitionList);
+    }
+
+    private void contextCreation(List<BeanDefinition> beanDefinitionList) {
+        log.debug("Context table generation");
+        for (BeanDefinition beanDefinition : beanDefinitionList) {
+            log.debug("Bean definition with name = {} and type = {} will be added", beanDefinition.name(), beanDefinition.type());
+            context.put(beanDefinition.name(), beanDefinition.type(), beanDefinition);
+        }
+    }
+
+    private void beansInstantiation(BeanInitializer initializer) {
+        log.info("Beans initialization");
+        initializer.initialize(context);
     }
 
-    @Override
-    public <T> void register(String name, Class<T> beanType) {
 
+    @Override
+    public <T> void register(String beanName, Class<T> beanType) {
+        // TODO:  add implementation
     }
 
     @Override
     public void register(Class<?>... beanTypes) {
-
+        // TODO:  add implementation
     }
 
     @Override
     public <T> String register(Class<T> beanType) {
         return null;
+        // TODO: add implementation
     }
 
     @Override
     public <T> T getBean(Class<T> beanType) {
-        return null;
+        Objects.requireNonNull(beanType, BEAN_TYPE_MUST_BE_NOT_NULL_MESSAGE);
+
+        List<T> beans = context.column(beanType)
+                .values()
+                .stream()
+                .map(beanDefinition -> beanType.cast(beanDefinition.instance()))
+                .toList();
+        if (beans.isEmpty()) {
+            throw new NoSuchBeanException(NO_SUCH_BEAN_EXCEPTION_MESSAGE.formatted(beanType.getSimpleName()));
+        } else if (beans.size() > 1) {
+            throw new NoUniqueBeanException("Expected single bean of type " + beanType.getSimpleName()
```

Comment:
You can use `format()` here - `String.format("Found %d instances of bean %s",  beans.size(), beanType.getSimpleName())`

Reply:
Done